### ISSUE TITLE
Code logic wierdness in experimental_paradigms.py

### ISSUE THREAD

**Issue by kchawla-pi** [nistats/nistats#252]
In `nistats/experimental_paradigms.py`, the code is a little weird, or could use a few comments explaining what's going on there.
First up:
Line # 54
```
    trial_type = np.repeat('dummy', n_events)
```
The trial_type is available in the events file so why are we generating this dummy data?

Second:
Line # 57 onwards
```
    if 'trial_type' in paradigm.keys():
        warnings.warn("'trial_type' key not found in the given paradigm.")
        trial_type = np.array(paradigm['trial_type'])
```
In effect we are generating the warning that the events data does not have the key 'trial_type' when it in fact does? And then we proceed to use that events data?
Same with subsequent code:
```
    if 'duration' in paradigm.keys():
        warnings.warn("'duration' key not found in the given paradigm.")
        duration = np.array(paradigm['duration']).astype(np.float)
    if 'modulation' in paradigm.keys():
        warnings.warn("'modulation' key not found in the given paradigm.")
        modulation = np.array(paradigm['modulation']).astype(np.float)
    return trial_type, onset, duration, modulation
```

**Comment by kchawla-pi** [comment]
Seems PR #236  is on top of this one.

**Comment by kchawla-pi** [comment]
Fixed in #236
